tests/providers/amazon/aws/operators/test_bedrock.py

```diff
@@ -346,3 +347,168 @@ def test_id_returned(self, mock_conn):
         result = self.operator.execute({})
 
         assert result == self.INGESTION_JOB_ID
+
+
+class TestBedrockRaGOperator:
+    VECTOR_SEARCH_CONFIG = {"filter": {"equals": {"key": "some key", "value": "some value"}}}
+    KNOWLEDGE_BASE_ID = "knowledge_base_id"
+    SOURCES = [{"sourceType": "S3", "s3Location": "bucket"}]
+    MODEL_ARN = "model arn"
+
+    @pytest.mark.parametrize(
+        "source_type, vector_search_config, knowledge_base_id, sources, expect_success",
+        [
+            pytest.param(
+                "invalid_source_type",
+                None,
+                None,
+                None,
+                False,
+                id="invalid_source_type",
+            ),
+            pytest.param(
+                "KNOWLEDGE_BASE",
+                VECTOR_SEARCH_CONFIG,
+                None,
+                None,
+                False,
+                id="KNOWLEDGE_BASE_without_knowledge_base_id_fails",
+            ),
+            pytest.param(
+                "KNOWLEDGE_BASE",
+                VECTOR_SEARCH_CONFIG,
+                KNOWLEDGE_BASE_ID,
+                SOURCES,
+                False,
+                id="KNOWLEDGE_BASE_with_sources_fails",
+            ),
+            pytest.param(
+                "KNOWLEDGE_BASE",
+                VECTOR_SEARCH_CONFIG,
+                KNOWLEDGE_BASE_ID,
+                None,
+                True,
+                id="KNOWLEDGE_BASE_passes",
+            ),
```

Comment:
Should another test case with just a knowledge_base_id (without vector config) also pass?

Reply:
Sure.   vector config is optional with a knowledge base, but forbidden when using external.  I can add that check.